### PR TITLE
[AC-5727] Add mocks for fetch calls

### PR DESCRIPTION
**Pull request by mc-nathan**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import JudgingRoundDisplay from './judging_round_display';
+import JudgingRoundSelector from './judging_round_selector';
+
+
+class App extends React.Component {
+    constructor(props) {
+	super(props);
+	this.state = {
+            judging_round: null,
```
Comment (MattMSumner) [inline src/app.js]: It looks like you have `transform-class-properties` setup in your [babelrc](https://github.com/masschallenge/application-allocator/blob/development/.babelrc#L3) so you can remove this constructor call and have:

```
class App extends React.Component {
  state = { judging_round: null }
...
```

This will be the default syntax in future react versions so that babel setting can eventually be removed. It's a nice change since it removes a good bit of boilerplate :)

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	};
+    }
+
+    select_judging_round = (_, data) => {
+	this.setState({judging_round: data.value})
+    }
+
+    render() {
+	return (<div>
+		<Header value='Application Allocator Setup'/>
```
Comment (MattMSumner) [inline src/app.js:17]: I'm not going to leave comments on style (apart from this one) but I will say that we've been using prettier on the directory and it's been a complete joy not having to make styling decisions on our code.

File: src/judging_round_display.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    componentDidUpdate(prevProps) {
+	const judging_round_id = this.props.judging_round
+	if (judging_round_id &&
+	      prevProps.judging_round !== this.props.judging_round) {
+	    this.fetch_data()
+	}
+    }
+
+    fetch_data() {
+	fetch("http://localhost:8000/api/v1/judging_round/" +
```
Comment (MattMSumner) [inline src/judging_round_display.js]: ES6 introduces template strings which gives the ability to interpolate values. You make them using backticks instead of quotes so you could write this as:

```js
fetch(`http://localhost:8000/api/v1/judging_round/${judging_round_id}/`
```

File: src/judging_round_display.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	      judging_round_id + "/",
+	      {credentials: "include", mode: "cors"})
+	    .then(res => res.json()).then(data => {
+		this.setState({ data: data })
```
Comment (MattMSumner) [inline src/judging_round_display.js]: When the keys and values are the same literal string, you can shorten this to:
```
this.setState({ data })
```

File: src/judging_round_display.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	    })
+    }
+
+    
+    render() {
+	if (this.props.judging_round) {
+	    return (<ul>
+		    <li>Props Judging Round: {this.props.judging_round}</li>
+		    <li>Full Name: {this.state.data.full_name}</li>
+		    <li>Id: {this.state.data.id}</li>
```
Comment (MattMSumner) [inline src/judging_round_display.js]: There's a lot of drilling into `this.state.data` throughout this render and you could pull some constants out using destructing above. Something like:

```js
render() {
  if (this.props.judging_round) {
    const { judging_round } = this.props
    const { full_name, id, is_active, round_type, cycle_based_round, program_id } = this.state.data
    return (
      <ul>
        <li>Props Judging Round: {judging_round}</li>
...etc...
    )
  } else {
    return (<div>No Judging Round Selected</div>)
  }
}
```

File: src/judging_round_display.test.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    it('says No Judging Round Selected', () => {
+	const display = shallow(<JudgingRoundDisplay judging_round={null}/>);
+	expect(display.text()).toContain('No Judging Round Selected');
+    });
+});
+
+
+describe('judging round display with judging round', () => {
+    beforeEach(() => {
+	global.fetch = jest.fn().mockImplementation(
```
Comment (MattMSumner) [inline src/judging_round_display.test.js:18]: This works well. We've also been trying out https://github.com/jefflau/jest-fetch-mock which has some nice test helpers for setting up these promises.

File: src/judging_round_selector.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
+import { Form, Select } from 'semantic-ui-react'
+
+
+class JudgingRoundSelector extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
```
Comment (MattMSumner) [inline src/judging_round_selector.js]: Same comment here about being able to remove some boiler plate. This could be:

```js
class JudgingRoundSelector extends React.Component {
  state = {
    error: null,
    isLoaded: false,
    user: []
  }

**Review by MattMSumner — APPROVED** [review]
Overall this looks good. I've left a lot of comments that I'll leave up to you to decide if they're worth addressing. I'd welcome your thoughts if you don't.

File: src/judging_round_display.js (hunk 3 above)
Comment (mc-nathan) [reply src/judging_round_display.js]: Addressed in AC-5696.

File: src/judging_round_display.js (hunk 4 above)
Comment (mc-nathan) [reply src/judging_round_display.js]: Done.

File: src/judging_round_selector.js (hunk 7 above)
Comment (mc-nathan) [reply src/judging_round_selector.js]: Done.

File: src/app.js (hunk 1 above)
Comment (mc-nathan) [reply src/app.js]: Done.

File: src/judging_round_display.js (hunk 5 above)
Comment (mc-nathan) [reply src/judging_round_display.js]: Didn't go quite as far as you suggest, but at least removed some of the extra drilling.

File: src/judging_round_display.test.js (hunk 6 above)
Comment (mc-nathan) [reply src/judging_round_display.test.js:18]: Not going to dig into that now, but will keep it in mind for later.

File: src/app.js (hunk 2 above)
Comment (mc-nathan) [reply src/app.js:17]: I see if I can get that hooked up with emacs.  Emacs has a JavaScript mode which is fairly opinionated and I'm not sure how well the two will play with each other.  Not going to fix for this PR though.
